app/services/attachments/destroy_service.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Attachments
+  # Service used to Delete Projects
+  class DestroyService < BaseService
+    def initialize(attachable, attachment, user = nil)
+      super(user)
+      @attachable = attachable
+      @attachment = attachment
+    end
+
+    def execute
+      authorize! @attachable.project, to: :destroy?
+
```

Comment:
Before deleting we should make sure that the attachment (and it's associated attachment if paired-end) belongs to the sample. If not an appropriate error should be attached to the `@attachment`. You can take a look at the other services to see examples of how the errors are attached

Reply:
Fixed in 4376c31d10cd19542a8c6f5a924ecc659f51f78c

Reply:
As mentioned below, fixed in 3f2b8e27a44b3a94983c26198cee222bb520f77e